metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/ScheduledWorkflowDao.java

```diff
@@ -206,13 +210,21 @@ public boolean deleteAllByDatasetId(String datasetId) {
             Date.from(lowerBound.atZone(ZoneId.systemDefault()).toInstant()))).and(
         query.criteria("pointerDate")
             .lessThan(Date.from(upperBound.atZone(ZoneId.systemDefault()).toInstant())));
-    query.order(ID.getFieldName());
-    return ExternalRequestUtil.retryableExternalRequestConnectionReset(() -> query.asList(
-        new FindOptions().skip(nextPage * getScheduledWorkflowPerRequest())
-            .limit(getScheduledWorkflowPerRequest())));
+    query.order(Sort.ascending(ID.getFieldName()));
+    final FindOptions findOptions = new FindOptions()
+        .skip(nextPage * getScheduledWorkflowPerRequest())
+        .limit(getScheduledWorkflowPerRequest());
+    return ExternalRequestUtil
+        .retryableExternalRequestConnectionReset(() -> getListOfQuery(query, findOptions));
 
   }
 
+  private <T> List<T> getListOfQuery(Query<T> query, FindOptions findOptions) {
+    try (MorphiaCursor<T> cursor = query.find(findOptions)) {
+      return cursor.toList();
+    }
+  }
```

Comment:
Since you're only calling this method from within a retryable connection, perhaps you can move the 'ExternalRequestUtil.retryableExternalRequestConnectionReset' to this private method as well? Same for the other DAO above.

Reply:
sure